### PR TITLE
test: allow overriding the worker fixtures

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: test/runner/runner.js [hunk 1]
```diff
@@ -63,11 +65,28 @@ class Runner extends EventEmitter {
     }
   }
 
+  _filesSortedByWorkerHash() {
+    const result = [];
+    for (const file of this._files.keys())
+      result.push({ file, hash: computeWorkerHash(file) });
+    result.sort((a, b) => a.hash < b.hash ? -1 : (a.hash === b.hash ? 0 : 1));
+    console.log(result);
```
Comment (JoelEinbinder) [inline test/runner/runner.js:73]: stray log
```suggestion
```

**Review by dgozman — APPROVED** [review]

File: test/runner/runner.js [hunk 2]
```diff
@@ -214,4 +232,30 @@ class Worker extends EventEmitter {
...
+  const deps = cache.children.map(m => m.id);
+  for (const dep of deps)
+    collectRequires(dep, allDeps);
+}
+
+function computeWorkerHash(file) {
+  // At this point, filesWithRegistrations contains all the files with worker fixture registrations.
+  // For every test, build the require closure and map each file to fixtures declared in it.
+  // This collection of fixtures is the fingerprint of the worker setup, a "worker hash".
+  // Tests with the matching "worker hash" will reuse the same worker.
```
Comment (JoelEinbinder) [inline test/runner/runner.js:249]: If I have a `server` worker fixture, and I add an unrelated `pokemon` worker fixture in my test, does this mean I will get a new server instance?

Comment (pavelfeldman) [reply test/runner/runner.js:249]: Yes, you will. I can trace fixture usages and overrides to mitigate this.

Comment (pavelfeldman) [reply test/runner/runner.js:249]: Having said that, I am not sure it is critical - all I did was I recycled your worker...

Comment (JoelEinbinder) [reply test/runner/runner.js:249]: We can do it later. Just trying to make sure I understand what im reading.

File: test/runner/runner.js (hunk 1 above)
Comment (pavelfeldman) [reply test/runner/runner.js]: Outdated patch :P

**Pull request merged**